CMakeLists.txt

```diff
@@ -408,6 +409,59 @@ target_link_directories(notcurses PRIVATE ${OIIO_LIBRARY_DIRS})
 target_link_directories(notcurses-static PRIVATE ${OIIO_STATIC_LIBRARY_DIRS})
 endif()
 
+#######################################
+# libnotcurses-ffi (ffi shared library)
+file(GLOB NCFFISRCS CONFIGURE_DEPENDS src/libffi/*.c src/libffi/*.cpp)
+add_library(notcurses-ffi SHARED ${NCFFISRCS})
+target_compile_options(notcurses-ffi PUBLIC -fkeep-inline-functions)
+target_compile_definitions(notcurses-ffi PUBLIC NOTCURSES_FFI)
+# don't want these on freebsd/dragonfly/osx
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+target_compile_definitions(notcurses-ffi
+  PUBLIC
+   _XOPEN_SOURCE=700 # wcwidth(3) requires _XOPEN_SOURCE, and is in our headers
+  PRIVATE
+   _GNU_SOURCE _DEFAULT_SOURCE
+)
+endif()
+set_target_properties(notcurses-ffi PROPERTIES
+  VERSION ${PROJECT_VERSION}
+  SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+target_include_directories(notcurses-ffi
+  BEFORE
+  PRIVATE
+    include
+    src
+    "${CMAKE_REQUIRED_INCLUDES}"
```

Comment:
This and the stuff below is copy pasta from `notcurses-core`.

Reply:
i might try to factor some of this out, but you don't have to